base/linalg/eigen.jl

```diff
@@ -27,6 +27,45 @@ function getindex(A::Union{Eigen,GeneralizedEigen}, d::Symbol)
     throw(KeyError(d))
 end
 
+eigsortby(λ::Real) = λ
+eigsortby(λ::Complex) = reim(λ)
+
+"""
+    sort!(F::Base.LinAlg.Eigen; kw...)
+
+Sort the eigenvectors and eigenvalues of an eigenfactorization `F` in place using the eigenvalues
+for comparison.  See also `sort!(::AbstractVector)`.
+
+# Examples
+```jldoctest
+julia> F = eigfact([4.0 1.0; 0 1.0])
+Base.LinAlg.Eigen{Float64,Float64,Array{Float64,2},Array{Float64,1}}([4.0, 1.0], [1.0 -0.316228; 0.0 0.948683])
+
+julia> F[:values]
+2-element Array{Float64,1}:
+ 4.0
+ 1.0
+
+julia> sort!(F)
+Base.LinAlg.Eigen{Float64,Float64,Array{Float64,2},Array{Float64,1}}([1.0, 4.0], [-0.316228 1.0; 0.948683 0.0])
+
+julia> F[:values]
+2-element Array{Float64,1}:
+ 1.0
+ 4.0
+```
+"""
+function sort!(F::Eigen; by=eigsortby, kw...)
+    if !issorted(F[:values], by=by, kw...)
+        perm = sortperm(F[:values]; by=by, kw...)
+        permute!(F[:values], perm)
+        Base.permutecols!!(F[:vectors], perm)
+    end
+    return F
+end
+
+sort(F::Eigen; kw...) = sort!(deepcopy(F), kw...)
```

Comment:
Could be worthwhile to define a `copy` method for `Eigen`. Such methods exists for e.g. `Cholesky` and `LU` factorizations. Something like:
```julia
Base.copy(F::Eigen) = Eigen(copy(F.values), copy(F.vectors))
```